sql/src/main/java/io/crate/planner/operators/Collect.java

```diff
@@ -135,6 +137,37 @@
         return preFetchSymbols;
     }
 
+    /**
+     * Return columns which are not used.
+     *
+     * Examples:
+     *
+     * <pre>
+     * toCollect: [f(x)]        used: x
+     * unused:    []
+     *
+     * toCollect: [x, f(x)]     used: [f(x)]
```

Comment:
Should used be `x` here?

Reply:
That was deliberate to show that `used` my also contain scalars. (Ex. in the case of `ORDER BY x + 1`)

Reply:
That makes sense. Still, wouldn't the algorithm output `[x, f(x)]`?

Reply:
no, used is empty in this case - I've added a test for this case

Reply:
>no, used is empty in this case

Why is the example then still saying `f(x)`?

Reply:
Sorry, I meant that, `unused` is empty